chrome/content/zotero/xpcom/quickCopy.js

```diff
@@ -25,27 +25,50 @@
 
 
 Zotero.QuickCopy = new function() {
-	this.getFormattedNameFromSetting = getFormattedNameFromSetting;
-	this.getSettingFromFormattedName = getSettingFromFormattedName;
-	this.getContentType = getContentType;
-	this.stripContentType = stripContentType;
-	this.getFormatFromURL = getFormatFromURL;
-	this.getContentFromItems = getContentFromItems;
 	
 	var _initialized = false;
 	var _formattedNames = {};
 	
+	/*
+	 * Return Quick Copy setting object from string
+	 */
+	this.unserializeSetting = function (setting) {
+		var settingObject = {};
+		
+		if (typeof setting === 'string' || setting instanceof String) {
+			try {
+				settingObject = JSON.parse(setting);
+			} catch (e) {
+				var parsedSetting = setting.match(/(bibliography|export)(?:\/([^=]+))?=(.+)$/);
+				if (parsedSetting) {
+					settingObject.mode = parsedSetting[1] ? parsedSetting[1] : '';
+					settingObject.contentType = parsedSetting[2] ? parsedSetting[2] : '';
+					settingObject.id = parsedSetting[3] ? parsedSetting[3] : '';
+					settingObject.locale = '';
+				}
+			}
+		} else {
+			// return setting if not a string, in which case it might already be an object
+			return setting;
```

Comment:
I think I actually depend on this function returning the original object if fed an object, but I'd have to check.